packages/aws-cdk/lib/api/cxapp/exec.ts

```diff
@@ -16,25 +16,21 @@ export async function execProgram(aws: SdkProvider, config: Configuration): Prom
   await populateDefaultEnvironmentIfNeeded(aws, env);
 
   const pathMetadata: boolean = config.settings.get(['pathMetadata']) ?? true;
-
   if (pathMetadata) {
     context[cxapi.PATH_METADATA_ENABLE_CONTEXT] = true;
   }
 
   const assetMetadata: boolean = config.settings.get(['assetMetadata']) ?? true;
-
   if (assetMetadata) {
     context[cxapi.ASSET_RESOURCE_METADATA_ENABLED_CONTEXT] = true;
   }
 
   const versionReporting: boolean = config.settings.get(['versionReporting']) ?? true;
-
-  if (!versionReporting) {
-    context[cxapi.DISABLE_VERSION_REPORTING] = true;
-  }
+  if (versionReporting) { context[cxapi.VERSION_REPORTING_ENABLED_CONTEXT] = true; }
+  // We need to keep on doing this for framework version from before this flag was deprecated.
+  if (!versionReporting) { context[cxapi.DISABLE_VERSION_REPORTING] = true; }
```

Comment:
I wonder if we can remove this constant from cxapi and just plug the explicit value here for backwards compat

Reply:
Oh please yes